File: Python/app/FileTransfer.py
import os
import os.path
from re import compile
from shutil import move

fileList = os.listdir(os.getcwd())
extentions = open("Extentions.txt","r")
pattern = list(extentions.readline().split(','))
dng_path = os.getcwd+"..\DNG"
jpg_path = os.getcwd+"..\DCIM"
for file in fileList:
    for value in pattern:
        ptr = compile("\."+value+"$")
        if ptr.search(file):
            if value=="DNG":
                new_path = move(os.getcwd+"\\"+f,dng_path)
            elif value=="JPG":
                new_path = move(os.getcwd+"\\"+f,jpg_path)
            break
extentions.close()

File: java/app/FileTransfer.java
package java.app;

import java.io.IOException;
import java.util.regex.Pattern;
import java.util.regex.Matcher;
import java.io.File;
import java.nio.file.FileSystems;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class FileTransfer{
    public static void main(String[] args) throws IOException{
        String path = System.getProperty("user.dir");
        File file = new File(path);
        File[] fileList = file.listFiles();
        String separator = FileSystems.getDefault().getSeparator();
        if (fileList!=null){
            for (int i=0;i<fileList.length;i++){
                File f = fileList[i];
                String str = f.getName();
                Pattern cp = Pattern.compile("\\.exe$");
                Matcher mp = cp.matcher(str);

                if (!mp.find()){
                    Pattern p = Pattern.compile("\\.DNG$");
                    Matcher m = p.matcher(str);

                    if (m.find()){
                        try{
                            Path sourcePath = Paths.get(path+separator+str);
                            Path targetPath = Paths.get(".."+separator+"DNG"+separator+str);
                            Files.move(sourcePath,targetPath);
                            System.out.println("DNG");
                        } catch (IOException e){
                            e.printStackTrace();
                        }
                    } else {
                        try{
                            Path sourcePath = Paths.get(path+separator+str);
                            Path targetPath = Paths.get(".."+separator+"DCIM"+separator+str);
                            Files.move(sourcePath, targetPath);
                            System.out.println("JPG");
                        } catch (IOException e){
                            e.printStackTrace();
                        }
                    }
                }
            }
        }
    }
}